lib/OpenQA/Worker/WebUIConnection.pm

```diff
@@ -314,9 +314,9 @@ sub send {
         $msg //= $err->{message};
         if (my $error_code = $err->{code}) {
             $msg = "$error_code response: $msg";
-            if ($error_code == 404) {
-                # don't retry on 404 errors (in this case we can't expect different
-                # results on further attempts)
+            if (400 <= $error_code && $error_code < 500) {
+                # don't retry on most 4xx errors (in this case we can't expect
```

Comment:
On *most*? It doesn't seem that the code makes any exceptions so far. The test also has a similarly misleading comment (unless I'm missing something). By the way, in my previous review I've mentioned that some possible exceptions but likely we can ignore them for our purposes.

Reply:
With the latest change I also handle 408 and 425 so the comment and test fit better. Unless you want me to not handle 408+425 in a special way then please see the alternative PR #2899